test_runner/src/main/kotlin/ftl/args/AndroidArgs.kt

```diff
@@ -91,24 +91,27 @@ AndroidArgs
     }
 }
 
+// Changes these based on type
 val AndroidArgs.isDontAutograntPermissions
     get() = !(grantPermissions.isNotNull() && (grantPermissions.equals("all")))
 
 val AndroidArgs.isInstrumentationTest
-    get() = appApk.isNotNull() &&
-            testApk.isNotNull() ||
-            additionalAppTestApks.isNotEmpty() &&
-            (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() })
+    get() = if (type != null) (type == Type.INSTRUMENTATION) else
+        (appApk.isNotNull() && testApk.isNotNull() || additionalAppTestApks.isNotEmpty() &&
+            (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() }))
 
 val AndroidArgs.isRoboTest
-    get() = appApk.isNotNull() &&
-            (roboDirectives.isNotEmpty() || roboScript.isNotNull())
+    get() = if (type != null) (type == Type.ROBO) else
+        (appApk.isNotNull() && (roboDirectives.isNotEmpty() || roboScript.isNotNull()))
 
 val AndroidArgs.isSanityRobo
     get() = appApk.isNotNull() &&
             testApk.isNull() &&
             roboScript.isNull() &&
             additionalAppTestApks.isEmpty()
 
+val AndroidArgs.isGameLoop
+    get() = if (type == null) false else (type == Type.GAMELOOP)
```

Comment:
I don't like `type` flag, it's redundant in 2/3 and only really, required for gameloop. Is there any easy way to identify gameloop without this flag? If not. skip this comment.

Reply:
No there isnt an easy way. The type flag appears to be what probably should have been used to identify the robo, instrumental and gameloop in the first place.